src/Pages/VideoListing/VideoListing.jsx

```diff
@@ -1,31 +1,110 @@
+import { useLocation, useSearchParams } from "react-router-dom";
+import { useEffect } from "react";
 import { VerticalCard } from "../../Components/Cards/VerticalCard";
 import FloatingButton from "../../Components/UI/Button/FloatingButton";
-import { useVideo } from "../../Context";
+import { useFilter, useVideo } from "../../Context";
+import { finalFilteredData } from "../../Utils/finalFilteredData";
 import "./VideoListing.css";
 
 export const VideoListing = () => {
   const {
     videoState: { videos, categories },
   } = useVideo();
+  const { filterState, filterDispatch } = useFilter();
+  const { byCategory, bySearch, byLatest } = filterState;
+  const { search } = useLocation();
+  const urlParam = new URLSearchParams(search);
+  const searchQuery = urlParam.get("query");
+
+  const onCategoryClickHandler = (category) => {
+    filterDispatch({ type: "FILTER_CATEGORY", payload: category });
+  };
+
+  const onAllClickHandler = () => {
+    filterDispatch({ type: "ALL_CATEGORY" });
+  };
+
+  const onLatestClickHandler = () => {
+    if (byLatest) {
+      filterDispatch({ type: "REMOVE_LATEST_VIDEOS" });
+    } else {
+      filterDispatch({ type: "LATEST_VIDEOS", payload: "latest" });
+    }
+  };
+
+  const videosAvailable =
+    finalFilteredData(videos, filterState).length > 0 ? true : false;
+
+  useEffect(() => {
+    if (searchQuery) {
+      filterDispatch({ type: "SEARCH_VIDEO", payload: searchQuery });
+      urlParam.delete("query");
+    } else {
+      filterDispatch({ type: "ALL_CATEGORY" });
+    }
+  }, [searchQuery]);
 
   return (
     <div className="video-listing-body">
       <div className="discover-section"></div>
       <form className="category-section">
-        <div className="category-label">All</div>
+        <div
+          onClick={onAllClickHandler}
```

Comment:
should'nt this be a callback function in onClick?

Reply:
If we are calling a declared function then we provide that function name. If we don't want to declare a function then we provide a callback function.